### PR TITLE
Documentation: Document how to transfer files from QEMU to the host

### PR DESCRIPTION
**Pull request by electrikmilk**
Draft of how-to transfer files from Serenity to your host machine.

After a conversation I had with Agni on SerenityOS Discord, and after looking around in the documentation, I noticed there doesn't appear to be any documentation on how to retrieve files from Serenity that can only be modified on the system (e.g. Fonts, etc) to commit from your host machine.

(This was a previous PR, but I could not figure out how to remove the unrelated commits)

### CODE REVIEW

**Comment by Xexxa** [comment]
"ws ." starts the webserver on the default port (8000), probably a bit easier to remember.

> NOTE: If you are trying to download a Serenity specific binary (eg. Katica.font), try using wget to download the binary instead.

I always right-click and download, have I been doing it wrong all this time? :-O

**Comment by electrikmilk** [comment]
@Xexxa I found when downloading a `.font` file, the browser doesn't recognize that file and downloads it as plain text, Serenity then sees this .font file that has a plain text header as corrupted. I ended up losing work because of this, so as to not let others befall the same fate, it might be good to note that they should use something that downloads the file AS IS.

**Comment by Xexxa** [comment]
> @Xexxa I found when downloading a `.font` file, the browser didn't recognize the file and downloaded it as plain text, Serenity then sees this `.font` file that has a plain text header as corrupted. I ended up losing work because of this, so as to not let others befall the same fate, it might be good to note that they should use something that downloads the file AS IS.

Good point. Firefox/Ubuntu seem to download them as binary so then I guess I don't have to worry then. Thank you for the information!

**Comment by electrikmilk** [comment]
@Xexxa Thank you for your suggestions, I updated it to use `ws .` shorthand instead, and word that note a bit better to explain the plain text issue with some browsers.

File: Documentation/TransferringFIles.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Transferring files from Serenity to your host machine
+
+## Method 1: WebServer
+Serenity has a built-in web server which extends to your host machine.
+
+Open a new terminal and use the following command to start a WebServer instance for the home directory:
+
+ ```
+ ws .
+ ```
```
Comment (linusg) [inline Documentation/TransferringFIles.md:10]: Why is this indented by one space? Also should have a code block language (`console`).

File: Documentation/TransferringFIles.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+## Method 2: Mount disk_image
+
+Another way is to mount Serenity's disk_image to your host machine by using the following command on *nix systems:
+
+ ```
+ cd Build/{YOUR_BUILD_DIRECTORY}
+ mkdir mnt
+ sudo mount -t ext2 _disk_image mnt
+ ```
```
Comment (linusg) [inline Documentation/TransferringFIles.md:26]: Same as above

File: Documentation/TransferringFIles.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+![](WebServer_localhost.png)
+
+**NOTE:** Due to the fact that same browsers download unrecognized files as plain text, you may want to use something like `wget` to download the file **as is** instead. Otherwise the file may appear corrupted when the system tries to load it. 
+
+## Method 2: Mount disk_image
+
+Another way is to mount Serenity's disk_image to your host machine by using the following command on *nix systems:
+
+ ```
+ cd Build/{YOUR_BUILD_DIRECTORY}
```
Comment (linusg) [inline Documentation/TransferringFIles.md:23]:
```suggestion
 cd "Build/${SERENITY_ARCH}"
```

File: Documentation/TransferringFIles.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# Transferring files from Serenity to your host machine
+
+## Method 1: WebServer
+Serenity has a built-in web server which extends to your host machine.
+
+Open a new terminal and use the following command to start a WebServer instance for the home directory:
```
Comment (linusg) [inline Documentation/TransferringFIles.md:6]:
```suggestion
Open a new terminal and use the following command to start a WebServer instance for the current working directory:
```

**Review by linusg — CHANGES_REQUESTED** [review]
Typo in the markdown document filename: `FIles`

Please call the commit message something like `Documentation: Document how to transfer files from QEMU to the host`

File: Documentation/TransferringFIles.md (hunk 1 above)
Comment (electrikmilk) [reply Documentation/TransferringFIles.md]: I didn't know `console` would be the correct code block language for this case, I'll change that.

As for the indentation, I generally feel it's easier to copy-paste if it's one line, but thinking about it now I don't know if the clipboard is generally carried over into QEMU. Either way, if this is not how Serenity documentation is written I am fine with that.

File: Documentation/TransferringFIles.md (hunk 2 above)
Comment (electrikmilk) [reply Documentation/TransferringFIles.md]: Will do

**Review by electrikmilk — COMMENTED** [review]
Requested changes make sense to me. Code block changes will be made in a second commit.

File: Documentation/TransferringFIles.md (hunk 1 above)
Comment (electrikmilk) [reply Documentation/TransferringFIles.md]: Where are you seeing an intent? Do you mean because this is 3 lines instead of just one?

Comment (linusg) [reply Documentation/TransferringFIles.md]: No, I mean what I wrote - it's indented by one space:

image

**Comment by linusg** [comment]
Not sure what the state of this PR is. I'm not merging this with the wrongly indented code blocks.

**Comment by electrikmilk** [comment]
> Not sure what the state of this PR is. I'm not merging this with the wrongly indented code blocks.

I have removed the indented code blocks, I think my editor must have done that for some reason.

**Review by linusg — COMMENTED** [review]
Please squash the fixup commit as per the [contribution guidelines](https://github.com/SerenityOS/serenity/blob/master/CONTRIBUTING.md#code-submission-policy) :)

**Pull request merged**